examples/mint-repl/mint-repl.rs

```diff
@@ -0,0 +1,1110 @@
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+//! Safe Network DBC Mint CLI playground.
+
+use anyhow::{anyhow, Error, Result};
+use rustyline::config::Configurer;
+use rustyline::error::ReadlineError;
+use rustyline::Editor;
+use serde::{Deserialize, Serialize};
+use sn_dbc::{
+    BlindedOwner, Dbc, DbcContent, DbcTransaction, Hash, KeyManager, Mint, MintSignatures,
+    NodeSignature, ReissueRequest, ReissueTransaction,
+};
+use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
+use std::iter::FromIterator;
+use threshold_crypto::poly::Poly;
+use threshold_crypto::serde_impl::SerdeSecret;
+use threshold_crypto::{
+    PublicKey, PublicKeySet, SecretKeySet, SecretKeyShare, Signature, SignatureShare,
+};
+
+/// Holds information about the Mint, which may be comprised
+/// of 1 or more nodes.
+struct MintInfo {
+    mintnodes: Vec<Mint>,
+    genesis: DbcOwned,
+    secret_key_set: SecretKeySet,
+    poly: Poly,
+}
+
+impl MintInfo {
+    // returns the first mint node.
+    fn mintnode(&self) -> Result<&Mint> {
+        self.mintnodes
+            .get(0)
+            .ok_or_else(|| anyhow!("Mint not yet created"))
+    }
+}
+
+/// A Dbc plus the owner's pubkey set
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+struct DbcOwned {
+    inner: Dbc,
+    owner: PublicKeySet,
+}
+
+/// A ReissueTransaction with pubkey set for all the input and output Dbcs
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct ReissueTransactionOwned {
+    inner: ReissueTransaction,
+    inputs_owners: HashMap<Hash, PublicKeySet>,
+    outputs_owners: HashMap<Hash, PublicKeySet>,
+}
+
+/// A ReissueRequest with pubkey set for all the input and output Dbcs
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct ReissueRequestOwned {
+    inner: ReissueRequest,
+    inputs_owners: HashMap<Hash, PublicKeySet>,
+    outputs_owners: HashMap<Hash, PublicKeySet>,
+}
+
+/// This type is just for serializing HashMap<Hash, <HashMap<usize, SignatureShare>>
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+struct SignatureSharesMap(HashMap<Hash, HashMap<usize, SignatureShare>>);
+
+/// program entry point and interactive command handler.
+fn main() -> Result<()> {
+    print_logo();
+    println!("Type 'help' to get started.\n");
+
+    let mut mintinfo: MintInfo = mk_new_random_mint(0, 1000)?;
+
+    let mut rl = Editor::<()>::new();
+    rl.set_auto_add_history(true);
+    'outer: loop {
+        match rl.readline(">> ") {
+            Ok(line) => {
+                let mut args = line.trim().split_whitespace();
+                let cmd = if let Some(cmd) = args.next() {
+                    cmd
+                } else {
+                    continue 'outer;
```

Comment:
Is this `outer` label still required? I'm not seeing an inner loop here

Reply:
good catch.  that came from qp2p example I started with, and I might have removed a loop somewhere in the process.  I'm a little surprise clippy doesn't check for that.   anyway, I'll change it.